lib/ansible/modules/cloud/vagrant/vagrant_box.py

```diff
@@ -0,0 +1,290 @@
+        self.provider = module.params.get('provider')
+        self.insecure = module.params.get('insecure')
+        self.trust_location = module.params.get('trust_location')
+        self.cacert = module.params.get('cacert')
+        self.capath = module.params.get('capath')
+        self.cert = module.params.get('cert')
+        self.name = module.params.get('name')
+        self.checksum = module.params.get('checksum')
+        self.checksum_type = module.params.get('checksum_type')
+
+    def ispresent(self):
+        cmd = "%s box list" % self.module.get_bin_path('vagrant', required=True)
+        rc, out, err = self.module.run_command(cmd)
+        if os.path.exists(self.box):
+            if self.name:
+                box = self.name
+            else:
+                box = os.path.basename(self.box)
+        else:
+            box = self.box
+        if self.version is not "latest":
+            result = re.search('%s[ \t]*\(%s, %s\)' % (box, self.provider, self.version), out)
+
+            if result:
+                return True
+            else:
+                return False
+
+    def add(self):
+        if self.ispresent() and not self.force:
+            self.module.exit_json(changed=False, stdout="The box: '%s' version: '%s' provider: '%s' already exists" % (self.box, self.version, self.provider))
+
+        opts_list = ['--clean']
+        opts_dict = dict()
+        # options only that DO NOT need a value
+        if self.force:
+            opts_list.append('--force')
+        if self.insecure:
+            opts_list.append('--insecure')
+        if self.trust_location:
+            opts_list.append('--location-trusted')
+
+        # options that NEED a value
+        opts_dict['--provider'] = self.provider
+        if self.cacert:
+            opts_dict['--cacert'] = self.cacert
+        if self.capath:
+            opts_dict['--capath'] = self.capath
+        if self.cert:
+            opts_dict['--cert'] = self.cert
+        if self.version is not "latest":
+            opts_dict['--box-version'] = self.version
+
+        # conditional options, they are added ONLY IF the box is a filename
+        if os.path.exists(self.box):
+            if self.name:
+                opts_dict['--name'] = self.name
+            else:
+                opts_dict['--name'] = os.path.basename(self.box)
+
+            if self.checksum:
+                opts_dict['--checksum'] = self.checksum
+                opts_dict['--checksum-type'] = self.checksum_type
+
+        if opts_dict:
+            for k, v in opts_dict.items():
+                opts_list.append(str(k) + " " + str(v))
+
+        cmd = "%s box add %s %s" % (self.module.get_bin_path('vagrant', required=True), self.box, " ".join(opts_list))
+        rc, out, err = self.module.run_command(cmd)
+
+        if rc == 0:
+            self.module.exit_json(changed=True, rc=rc, stdout=out, stderr=err)
+        elif "already exists" in err:
```

Comment:
When doing screen scraping, we have to make sure the command run in an expected LANG, see comment below